docs/pull_request_template.md

```diff
@@ -1,17 +1,31 @@
-Fixes P4-<TODO>
+### Jira link
 
-## Proposed Changes
+P4-<TODO>
 
-- TODO
+### What?
 
-## To test/demo change
+I have added/removed/altered:
 
-- TODO
+- [ ] Added foo in bar
+- [ ] Removed baz
 
-## Things to check and link
-- [ ] API docs has been updated if necessary
-- [ ] New environment variables have been added to staging configuration
+### Why?
 
```

Comment:
We should keep this as reminder, we tend to forget things, in case we don't need it in our PR we can manually remove it.